integrated_channels/blackboard/client.py

```diff
@@ -38,3 +43,104 @@ def create_course_completion(self, user_id, payload):  # pylint: disable=unused-
 
     def delete_course_completion(self, user_id, payload):  # pylint: disable=unused-argument
         """TODO"""
+
+    def _create_session(self):
+        """
+        Will only create a new session if token expiry has been reached
+        """
+        now = datetime.utcnow()
+        if self.session is None or self.expires_at is None or now >= self.expires_at:
+            # need new session if session expired, or not initialized
+            if self.session:
+                self.session.close()
+            # Create a new session with a valid token
+            oauth_access_token, expires_in = self._get_oauth_access_token(
+                self.enterprise_configuration.client_id,
+                self.enterprise_configuration.client_secret,
+            )
+            session = requests.Session()
+            session.headers['Authorization'] = 'Bearer {}'.format(oauth_access_token)
+            session.headers['Content-Type'] = 'application/json'
+            self.session = session
+            # expiry expected after `expires_in` seconds
+            if expires_in is not None:
+                self.expires_at = datetime.utcnow() + timedelta(seconds=expires_in)
+
+    def _get_oauth_access_token(self, client_id, client_secret):
```

Comment:
a bit embarrassed right now on the amount of copy pasting between this and canvas, but no worries because once this works well, I will be doing a refactor ticket, where I will also handle some tests for the client, so won't be going heavy on client test in this pr